src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3421,6 +3444,12 @@ public override BoundNode VisitConversion(BoundConversion node)
 
             (BoundExpression operand, Conversion conversion) = RemoveConversion(node, includeExplicitConversions: true);
             TypeSymbolWithAnnotations operandType = VisitRvalueWithResult(operand);
+
+            if (node.ConversionKind == ConversionKind.ExplicitNullable)
```

Comment:
>if (node.ConversionKind == ConversionKind.ExplicitNullable) [](start = 12, length = 59)

Is there a reason to not handle ```ImplicitNullable``` here as well?

Reply:
`VisitConversion` should only be called for explicit conversions.

---
In reply to: [250789805](https://github.com/dotnet/roslyn/pull/32729#discussion_r250789805) [](ancestors = 250789805)

Reply:
> VisitConversion should only be called for explicit conversions.

Can we assert that? Also, I do not think conversion classification should be affected by the fact whether conversion is explicitly applied in code. For example, int -> int? should be an implicit conversion even if a cast operator used in source code.

---
In reply to: [250790756](https://github.com/dotnet/roslyn/pull/32729#discussion_r250790756) [](ancestors = 250790756,250789805)